common/network-common/src/main/java/org/apache/spark/network/crypto/README.md

```diff
@@ -1,6 +1,20 @@
-Forward Secure Auth Protocol
+Forward Secure Auth Protocol v1.1
 ==============================================
 
+Deprecation Notice
+------------------
+This is a bespoke key exchange protocol that was implemented before Spark supported TLS (aka SSL) for RPC
+calls. It is recommended that Spark users upgrade to using TLS for RPC calls between Spark processes. This protocol
+will be deprecated and removed in the long-term.
+
+See
+the [Spark security documentation](https://github.com/apache/spark/blob/master/docs/security.md#ssl-encryption) for
+more information on how to configure TLS.
```

Comment:
I am not sure if we should announce deprecation given TLS is going to come out only in 4.0 ... we can recommend users to move to TLS though.

Would like feedback from others on this though ... @dongjoon-hyun, @srowen, @JoshRosen

Reply:
Given that SASL support still exists for backward compatibility, I am skeptical that this will actually be removed. I'll delete that comment.